### ISSUE TITLE
List object defined higher in the configuration hierarchy only partially overrides the same item defined lower in the hierarchy if the higher definition has fewer items

### ISSUE THREAD

**Issue by maddy-markovitz** [spring-projects/spring-boot#5696]
Say for example I have two configuration files, application.yml and application-development.yml.  They both contain a configuration item **myApp.myPropsList**, which is defined as a List of Strings.

application.yml:
```
myApp.myPropsList:
    - myProp1
    - myProp2
    - myProp3
```
application-development.yml:
```
myApp.myPropsList:
   - myProp4
   - myProp5
```
Expected behaviour:

 If I start my app with the development profile, I thought this would result in the list being loaded with {myProp4, myProp5}.  But actually it gets loaded with {myProp4, myProp5, myProp3}.  
I can see from the trace logging that this appears to be happening because Spring doesn't know where the list ends, so after finding myProp4 and myProp5 it continues to look for myApp.myPropsList[2], and it searches the whole configuration hierarchy, including application.yml.

Let me know if you require a sample app to demonstrate this.

**EDIT**
Looks like this is definitely a duplicate.  I'm using Spring Boot 1.2.3 - can you confirm if this has been fixed in a later release?

**Comment by maddy-markovitz** [comment]
Looks like this is related to https://github.com/spring-projects/spring-boot/issues/4313

**Comment by maddy-markovitz** [comment]
Apologies, I should have searched for this issue more thoroughly.  Looks like it's been fixed in 1.2.6.  Closing the ticket.
